verystable/bip32.py

```diff
@@ -0,0 +1,281 @@
+# Copyright (c) 2021 Anthony Towns
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test-only bip32 implementation"""
+
+import csv
+import hashlib
+import hmac
+import os
+import random
+import struct
+import unittest
+
+from .core import secp256k1 as SECP256K1
+from .core.key import ECKey, ECPubKey
+from .core.script import hash256, hash160
+from .core.util import assert_equal
+
+chars = '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
+def byte_to_base58(b):
+    b = b + hash256(b)[:4]
+    result = ''
+    value = int('0x' + b.hex(), 0)
+    while value > 0:
+        result = chars[value % 58] + result
+        value //= 58
+    while (b[0] == b'\0'):
+        result = chars[0] + result
+        b = b[1:]
+    return result
+
+def base58_to_byte(s):
+    """Converts a base58-encoded string to its data"""
+    if not s:
+        return b''
+    n = 0
+    for c in s:
+        n *= 58
+        assert c in chars
+        digit = chars.index(c)
+        n += digit
+    h = '%x' % n
+    if len(h) % 2:
+        h = '0' + h
+    res = n.to_bytes((n.bit_length() + 7) // 8, 'big')
+    pad = 0
+    for c in s:
+        if c == chars[0]:
+            pad += 1
+        else:
+            break
+    res = b'\x00' * pad + res
+
+    # Assert if the checksum is invalid
+    assert_equal(hash256(res[:-4])[:4], res[-4:])
+
+    return res[:-4]
```

Comment:
I think the base58 functions are available for import from `.core.address`.

Reply:
Yeah; I think I couldn't use that because of the `version` prefix that gets automatically added/removed.